ui/src/dashboards/components/NoteEditorOverlay.tsx

```diff
@@ -91,53 +87,56 @@ class NoteEditorOverlay extends PureComponent<Props, State> {
   }
 
   public render() {
-    const {
-      onDismiss,
-      params: {dashboardID},
-    } = this.props
+    const {dashboardID} = this.props
 
     if (!dashboardID) {
       return (
-        <Overlay visible={true}>
-          <Overlay.Container maxWidth={360}>
-            <Overlay.Header title="Oh no!" onDismiss={onDismiss} />
-            <Overlay.Body>
-              <h5>
-                This page does not allow creation or editing of notes, better
-                head to a dashboard to do that.
-              </h5>
-            </Overlay.Body>
-          </Overlay.Container>
-        </Overlay>
+        <Overlay.Container maxWidth={360}>
+          <Overlay.Header title="Oh no!" onDismiss={this.handleDismiss} />
+          <Overlay.Body>
+            <h5>
+              This page does not allow creation or editing of notes, better head{' '}
+              to a dashboard to do that.
+            </h5>
+          </Overlay.Body>
+        </Overlay.Container>
       )
     }
 
     return (
-      <Overlay visible={true}>
-        <Overlay.Container maxWidth={900}>
-          <Overlay.Header title={this.overlayTitle} onDismiss={onDismiss} />
-          <Overlay.Body>
-            <SpinnerContainer
-              loading={this.props.viewsStatus}
-              spinnerComponent={<TechnoSpinner />}
-            >
-              <NoteEditor />
-            </SpinnerContainer>
-          </Overlay.Body>
-          <Overlay.Footer>
-            <Button text="Cancel" onClick={onDismiss} />
-            <Button
-              text="Save"
-              color={ComponentColor.Success}
-              status={this.saveButtonStatus}
-              onClick={this.handleSave}
-            />
-          </Overlay.Footer>
-        </Overlay.Container>
-      </Overlay>
+      <Overlay.Container maxWidth={900}>
+        <Overlay.Header
+          title={this.overlayTitle}
+          onDismiss={this.handleDismiss}
+        />
+        <Overlay.Body>
+          <SpinnerContainer
+            loading={this.props.viewsStatus}
+            spinnerComponent={<TechnoSpinner />}
+          >
+            <NoteEditor />
+          </SpinnerContainer>
+        </Overlay.Body>
+        <Overlay.Footer>
+          <Button text="Cancel" onClick={this.handleDismiss} />
+          <Button
+            text="Save"
+            color={ComponentColor.Success}
+            status={this.saveButtonStatus}
+            onClick={this.handleSave}
+          />
+        </Overlay.Footer>
+      </Overlay.Container>
     )
   }
 
+  private handleDismiss = (): void => {
+    const {onDismiss, router} = this.props
+
+    onDismiss()
+    router.goBack()
```

Comment:
what does this do?

Reply:
`onDismiss()` just hides the overlay by removing stuff from redux
`router.goBack()` returns the router to the previous state for cleanliness purposes

Wasn't sure if these should be combined or not, since not all overlays are tied to the router